sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestConsumer.java

```diff
@@ -35,27 +40,57 @@ public class TestConsumer {
     @RegisterExtension
     public JdbiExtension h2Extension = JdbiExtension.h2().withInitializer(TestingInitializers.something()).withPlugin(new SqlObjectPlugin());
 
-    @Test
-    public void testReturnStream() {
-        Something one = new Something(3, "foo");
-        Something two = new Something(4, "bar");
-        Something thr = new Something(5, "baz");
+    Spiffy dao;
+    List<Something> expected;
+
+    @BeforeEach
+    public void setup() {
+        final Something one = new Something(3, "foo");
+        final Something two = new Something(4, "bar");
+        final Something thr = new Something(5, "baz");
 
-        Spiffy dao = h2Extension.getSharedHandle().attach(Spiffy.class);
+        dao = h2Extension.getSharedHandle().attach(Spiffy.class);
         dao.insert(one);
         dao.insert(thr);
         dao.insert(two);
 
-        List<Something> results = new ArrayList<>();
+        expected = Arrays.asList(thr, two, one);
+    }
+
+    @Test
+    public void consumeEach() {
+        final List<Something> results = new ArrayList<>();
         dao.forEach(results::add);
-        assertThat(results).containsExactly(thr, two, one);
+        assertThat(results).containsExactlyElementsOf(expected);
+    }
+
+    @Test
+    public void consumeIter() {
+        final List<Something> results = new ArrayList<>();
+        dao.consumeIter(iter -> iter.forEachRemaining(results::add));
+        assertThat(results).containsExactlyElementsOf(expected);
+    }
+
+    @Test
+    public void consumeStream() {
+        final List<Something> results = new ArrayList<>();
+        dao.consumeStream(stream -> stream.forEach(results::add));
+        assertThat(results).containsExactlyElementsOf(expected);
     }
 
     public interface Spiffy {
         @SqlQuery("select id, name from something order by id desc")
         @UseRowMapper(SomethingMapper.class)
         void forEach(Consumer<Something> consumer);
 
+        @SqlQuery("select id, name from something order by id desc")
+        @RegisterRowMapper(SomethingMapper.class)
+        void consumeIter(Consumer<Iterator<Something>> consumer);
```

Comment:
`consumeIterator`. It would be ironic to put "please use speaking names" in our contribution docs and the same day add a "iter" to the code base.